zokrates_fs_resolver/src/lib.rs

```diff
@@ -18,7 +20,18 @@ fn resolve_with_location(
     location: &String,
     source: &String,
 ) -> Result<(BufReader<File>, String, String), io::Error> {
-    let path = PathBuf::from(location).join(PathBuf::from(source));
+    let source = PathBuf::from(source);
+
+    // paths starting with `./` or `../` are interpreted relative to the current file
+    // other paths `abc/def.code` are interpreted relative to $ZOKRATES_HOME
+    let base = match source.components().next() {
+        Some(Component::CurDir) | Some(Component::ParentDir) => PathBuf::from(location),
```

Comment:
I'm not sure if this expected behavior, but with this implementation one can still import from  a the grandparent directory via this hack: `import "./../../test.code"` instead of using the not supported `.../test.code`.

Reply:
I think it's fine to go anywhere relatively.   As you said it would be a headache to prevent that and I don't really see a reason why. You could always symlink etc... File permissions should cover that not us. The implem here is just how paths are defined in rust: `.` and `..` are first class citizens.